course-definition.yml

```diff
@@ -676,4 +687,96 @@ stages:
     primary_extension_slug: "pub-sub"
     name: "Unsubscribe"
     difficulty: medium
-    marketing_md: In this stage, you'll add support for the `UNSUBSCRIBE command`, which is used to unsubscribe from a channel.
\ No newline at end of file
+    marketing_md: In this stage, you'll add support for the `UNSUBSCRIBE command`, which is used to unsubscribe from a channel.
+
+  # Sorted Sets
+  - slug: "ct1"
+    primary_extension_slug: "zsets"
+    name: "Create sorted set"
+    difficulty: easy
+    marketing_md: |
+      In this stage, you'll add support for creating a new sorted set using the `ZADD` command.
+
+  - slug: "hl2"
+    primary_extension_slug: "zsets"
+    name: "Validate score values"
+    difficulty: easy
+    marketing_md: |
+      In this stage, you'll add support for verifying the value of the score argument for `ZADD` command.
+
+  - slug: "hf1"
+    primary_extension_slug: "zsets"
+    name: "Add to existing zset"
+    difficulty: easy
+    marketing_md: |
+      In this stage, you'll add support for adding elements to an existing zset.
+
+  - slug: "kn4"
+    primary_extension_slug: "zsets"
+    name: "Count zset members"
```

Comment:
One thing I worry about with doing things like count, retrieve score etc. is that we're delaying enforcing a data structure - only in the rank stage do we actually test if a user is using a "sortable" data structure. 

Let's try to get to the core thing as fast as possible, and I think the best way to test that is `ZRANGE`. Let's place the ZRANGE stages first (List members, List with negative indexes). Since listing is by rank we can properly test all combinations of sorting logic (differnt scores, same score different lexographic key etc.).